deepchecks/vision/vision_data/format_validators.py

```diff
@@ -0,0 +1,208 @@
+                                  'where H and W are the height and width of the corresponding image.')
+        if np.max(single_image_label % 1) > 0:
+            raise ValidationError('In semantic segmentation, each pixel in label should represent a '
+                                  'class_id and there must be a positive integer.')
+
+
+def validate_predictions_format(predictions, task_type: TaskType):
+    """Validate that the predictions are in the required format based on task_type.
+
+    Parameters
+    ----------
+    predictions
+        The predictions of the batch. Result of VisionData's batch_to_predictions
+    task_type: TaskType
+        The task type of the model
+    Raises
+    ------
+    DeepchecksValueError
+        If the predictions doesn't fit the required deepchecks format.
+    """
+    single_image_pred = _validate_predictions_label_common_format('prediction', predictions, task_type)
+
+    if task_type == TaskType.CLASSIFICATION and single_image_pred is not None:
+        if not isinstance(single_image_pred, np.ndarray) or not isinstance(single_image_pred[0], Number) or \
+                not 0.99 < np.sum(single_image_pred) < 1.01:
+            raise ValidationError('Classification prediction per image must be a sequence of floats representing '
+                                  'probabilities per class.')
+    elif task_type == TaskType.OBJECT_DETECTION and single_image_pred is not None:
+        error_msg = 'Object detection prediction per image must be a sequence of 2D arrays, where each row ' \
+                    'has 6 columns: [x_min, y_min, w, h, confidence, class_id]'
+        if single_image_pred.shape[1] != 6:
+            raise ValidationError(error_msg)
+        if np.min(single_image_pred) < 0:
+            raise ValidationError(f'Found one of coordinates to be negative, {error_msg}')
+        if np.max(single_image_pred[:, 5] % 1) > 0:
+            raise ValidationError(f'Class_id must be a positive integer. {error_msg}')
+    elif task_type == TaskType.SEMANTIC_SEGMENTATION and single_image_pred is not None:
+        if len(single_image_pred.shape) != 3:
+            raise ValidationError('Semantic segmentation prediction per image must be a 3D array of shape (C, H, W),'
+                                  'where H and W are the height and width of the corresponding image, and C is the '
+                                  'number of classes that can be detected.')
+        if not 0.99 < np.sum(single_image_pred[:, 0][:, 0]) < 1.01:
+            raise ValidationError('Semantic segmentation prediction per pixel per image should be probabilities per '
+                                  'each possible class')
+
+
+def _validate_predictions_label_common_format(name, data, task_type: TaskType):
+    """Validate that the data is in the required format and returns a non-empty sample in numpy format."""
+    name_plural = name + 's'
```

Comment:
very weird especially when it is a private function where you pass the name yourself

Reply:
name is just for the readability of the exception. Saves a lot of duplicate code

Reply:
why the s?